src/rerank/rerank_calculator_ov.cc

```diff
@@ -153,7 +153,23 @@ class RerankCalculatorOV : public CalculatorBase {
         // Validate batch size before tokenizing
         if (handler.getDocumentsList().size() > this->max_allowed_chunks)
             throw std::runtime_error("Number of documents exceeds max_allowed_chunks");
-
+        if (!rerank_session->addBosToken) {
+            auto batchSize = handler.getDocumentsList().size();
+            std::vector<std::string> data(batchSize);
+            for (int i = 0; i < batchSize; i++) {
+                data[i] += handler.getQuery() + handler.getDocumentsList()[i];
+            }
+            chunk_mapping.resize(batchSize);
+            std::iota(chunk_mapping.begin(), chunk_mapping.end(), 0);
+            auto tokens = rerank_session->getTokenizer().encode(data);
+            if (tokens.input_ids.get_shape().size() != 2) {
+                throw std::runtime_error("Tokens shape invalid.");  // should never happen
+            }
+            if (this->max_position_embeddings < tokens.input_ids.get_shape()[1]) {
+                throw std::runtime_error("Query tokens count tokens count of longest document exceeds max_position_embeddings. For this model chunking is not supported.");
```

Comment:
The requests length of <tokens.input_ids.get_shape()[1]> tokens exceeds the model context of <max_position_embeddings>